sample_cgfs/rv32i_priv.cgf

```diff
@@ -1,21 +1,413 @@
 ecall:
   config: 
     - check ISA:=regex(.*I.*); def rvtest_mtrap_routine=True 
-  opcode: 
+  mnemonics: 
     ecall: 0
 
 ebreak:
   config: 
     - check ISA:=regex(.*I.*); def rvtest_mtrap_routine=True 
-  opcode: 
+  mnemonics: 
     ebreak: 0
+    
+ 
+misa_check_combination:
+  config: 
+    - check ISA:=regex(.*I.*Zicsr.*F.*D); 
+  csr_comb:
+    (misa & 0x08) | (misa & 0x20) == 0 : 0			# setting “F”=0 and “D”=1 should results in both “F” and “D” being cleared.
+
+
+misa_mxl_check:
+  config: 
+    - check ISA:=regex(.*I.*Zicsr.*); 
+  csr_comb:
+    misa >> (xlen-2) == 0x01 : 0				# A coverpoint which checks whether the mxl field of misa register is 1.
+
+
+mstatus_Smode:    
+  config: 
+    - check ISA:=regex(.*I.*Zicsr.*); 
+  mnemonics: 
+    csrrw: 0
+    csrrs: 0
+  csr_comb:
+    (mstatus >> 1) & 0x01 == 0x01 : 0				# checking if the SIE gets enabled
+    (mstatus >> 5) & 0x01 == 0x01 : 0 			# checking SPIE bit 
+    (mstatus >> 8) & 0x01 == 0x01 : 0 			# checking SPP bit
+    (mstatus >> 22) & 0x1 == 0x01 : 0 			# When TSR=1, attempts to execute SRET while executing in S-mode will raise an illegal instruction exception.  
+    (mstatus >> 36) & 0x1 == 0x01 : 0 			# check SBE = 1, it controls whether explicit load and store memory accesses made from S-mode is big-endian  
+    
+    
+
+mstatus_Mmode:    
+  config: 
+    - check ISA:=regex(.*I.*Zicsr.*); 
+  mnemonics: 
+    csrrw: 0
+    csrrs: 0
+  csr_comb:
+    (mstatus >> 3) & 0x01 == 0x01 : 0 			# checking if the MIE gets enabled
+    (mstatus >> 7) & 0x01 == 0x01 : 0 			# MPIE holds the value of the interrupt-enable bit active prior to the trap, checking whether it does that or not
+    (mstatus >> 10) & 0x3 == 0x3 : 0				# The xPP fields can only hold privilege modes up to x, so MPP is two bits wide, checking when if it becomes 11
+    (mstatus >> 10) & 0x2 == 0x2 : 0				# The xPP fields can only hold privilege modes up to x, so MPP is two bits wide, checking when if it becomes 10  
```

Comment:
Assuming this is meant to check if supervisor mode can be written to mpp the logic should be : 

`(mstatus >> 10) & 0x3 == 0x1`

this coverpoints should be shifted to mstatus_Smode coverage label since this value can only occur when supervisor mode is enabled.

Reply:
Updated